freezing/sync/data/activity.py

```diff
@@ -558,7 +608,36 @@ def is_excluded(activity):
             )
         ]
 
-        return filtered_rides
+        # If this rider has overlapping rides, just select the largest ride. This is because when
+        # we run a full sync the database may be empty so we pull all rides from Strava then filter
+        # and then insert, so filtering can't look at the database.
+        def overlaps_larger(activity, activities):
```

Comment:
This occurs when we do a full rider sync, so we've pulled all activities and will then insert them in bulk.